### PR TITLE
autoupdate pipeline, tracing, errexit

### PR DESCRIPTION
**Pull request by akinnane**

### CODE REVIEW

File: concourse/pipeline.yml [hunk 1]
```diff
@@ -54,15 +55,29 @@ resources:
...
+    serial: true
+    plan:
+      - get: cyber-security-nessus-git
+        trigger: true
+      - set_pipeline: nessus
+        file: cyber-security-nessus-git/concourse/pipeline.yml
+    on_failure:
+      <<: *health_status_notify
+      params:
+        message: "Cyber Zendesk Tickets pipeline update failed"
```
Comment (akinnane) [inline concourse/pipeline.yml:68]:
```suggestion
        message: "Cyber Security Nessus - pipeline update failed - https://github.com/alphagov/cyber-security-nessus/blob/master/concourse/pipeline.yml https://cd.gds-reliability.engineering/teams/cybersecurity-tools/pipelines/nessus"
```

**Review by denizgenc — COMMENTED** [review]
Changes themselves look fine, but I've noticed that the trigger for the `terraform_validation` job has no `passed` parameter, so from what I can tell it will run parallel to the `update-pipeline` job. I'm not sure what sort of effect that would have to the job that follows (`nessus_deploy`), especially since its trigger depends both on `terraform_validation` and `python_validation`.

File: concourse/pipeline.yml [hunk 2]
```diff
@@ -201,6 +223,9 @@ jobs:
             args:
               - -c
               - |
+                set -x
```
Comment (akinnane) [inline concourse/pipeline.yml:226]:
```suggestion
```

File: concourse/pipeline.yml [hunk 3]
```diff
@@ -292,6 +320,9 @@ jobs:
...
+                set -e
+
                 source /usr/local/bin/sts-assume-role.sh ((role_arn)) 'eu-west-2'
                 pip install pipenv
```
Comment (akinnane) [inline concourse/pipeline.yml:326]:
```suggestion

                set -e

                source /usr/local/bin/sts-assume-role.sh ((role_arn)) 'eu-west-2'
                set -x
                pip install pipenv
```

**Comment by akinnane** [comment]
> Changes themselves look fine, but I've noticed that the trigger for the `terraform_validation` job has no `passed` parameter, so from what I can tell it will run parallel to the `update-pipeline` job. I'm not sure what sort of effect that would have to the job that follows (`nessus_deploy`), especially since its trigger depends both on `terraform_validation` and `python_validation`. To be on the safe side I'd prefer to have the `terraform_validation` job depend wait for `update-pipeline` to pass.

I've updated the `terraform_validation` job to depend on the pipeline autoupdate stage.

**Review by denizgenc — APPROVED** [review]
Looks good!

**Pull request merged**